myapp/spec/views/tasks/edit.html.erb_spec.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'tasks/edit', type: :view do
+  let!(:task) { create(:task) }
+
+  before(:each) do
+    @task = assign(:task, task)
```

Comment:
Specのインスタンス変数にする必要は無さそう？
```suggestion
    assign(:task, task)
```

Reply:
ここは15行目のtask_path(@task)で使用しているのでテストしたインスタンス変数を使用して、
edit_pathの引数に引き渡すのはいかがでしょうか。。？

```ruby
let!(:task) { create(:task) }
let!(:assigned_task) { assign(:task, task) }

it 'renders the edit task form' do
  render

  assert_select 'form[action=?][method=?]', task_path(assigned_task), 'post' do
    assert_select 'input[name=?]', 'task[name]'
    assert_select 'select[name=?]', 'task[priority]'
    assert_select 'select[name=?]', 'task[status]'
    assert_select 'textarea[name=?]', 'task[explanation]'
  end
end
```

Reply:
それだと `task` と `assigned_task` は同じものですよね？
`task_path(task)` で行けませんか？

Reply:
同じですね。。なのでbefore(:each) do〜end周辺は削除致します！
```
[2] pry(#<RSpec::ExampleGroups::TasksEdit>)> task.object_id
=> 26020
[3] pry(#<RSpec::ExampleGroups::TasksEdit>)> assigned_task.object_id
=> 26020
```

Reply:
object_idは同じですがテンプレートを呼びだす(<%= render "form", task: @task %>)際にassignまたはインスタンス変数の使用を行う必要がありそうです。。
```ruby
[2] pry(#<RSpec::ExampleGroups::TasksEdit>)> task.object_id
=> 26020
[3] pry(#<RSpec::ExampleGroups::TasksEdit>)> assigned_task.object_id
=> 26020
```
コケました
```ruby
RSpec.describe 'tasks/edit', type: :view do
  let!(:task) { create(:task) }
  # let!(:assigned_task) { assign(:task, task) }

  it 'renders the edit task form' do
    render

    assert_select 'form[action=?][method=?]', task_path(task), 'post' do
      assert_select 'input[name=?]', 'task[name]'
      assert_select 'select[name=?]', 'task[priority]'
      assert_select 'select[name=?]', 'task[status]'
      assert_select 'textarea[name=?]', 'task[explanation]'
    end
  end
end
```
rspec
```ruby
Failures:

  1) tasks/edit renders the edit task form
     Failure/Error: <% if task.errors.any? %>

     ActionView::Template::Error:
       undefined method `errors' for nil:NilClass

       '.freeze;   if task.errors.any?
~
```

Reply:
アサインは必要ですが、インスタンス変数に代入する必要はないのでは？という指摘でした
↓これで行けると思うのですが
```ruby
RSpec.describe 'tasks/edit', type: :view do
  let!(:task) { create(:task) }

  before(:each) do
    assign(:task, task)
  end

  it 'renders the edit task form' do
    render

    assert_select 'form[action=?][method=?]', task_path(task), 'post' do
    # 略
```

Reply:
すみません。おっしゃる通りです！
自分の理解が浅かったです。。

Reply:
もし、`assign` の戻り値を捨てているのが気になるなら `let!(:task) { assign(:task, create(:task)) }` （`let!` である必要がある）として before ブロックを削除するという手もありそう。
ただ、個人的には現在の形のままのほうがやっていることがわかりやすいかなと思います。
（`assign` に戻り値があると思ってなかったｗ）